### PR TITLE
Move more matrix stuff to the right places

### PR DESCRIPTION
**Pull request by lgoettgens**

### CODE REVIEW

File: src/julia/Matrix.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#
+###############################################################################
+
+nrows(A::Matrix{T}) where {T} = size(A)[1]
+ncols(A::Matrix{T}) where {T} = size(A)[2]
+
+function is_zero_row(M::Matrix, i::Int)
+    for j = 1:ncols(M)
+        if !iszero(M[i, j])
+            return false
```
Comment (fingolfin) [inline src/julia/Matrix.jl:13]: There seem to be no tests for all this code. I wonder if Nemo or Hecke have any left-over test code?

File: src/julia/Matrix.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Convert `A` to a Julia `Matrix` of the same dimensions with the same elements.
+
+# Examples
+```jldoctest; setup = :(using AbstractAlgebra)
+julia> R, x = ZZ["x"]; A = R[x^0 x^1; x^2 x^3]
+[  1     x]
+[x^2   x^3]
+
+julia> Array(A)
```
Comment (fingolfin) [inline src/julia/Matrix.jl:59]: Should this perhaps eventually be changed from `Array` to `Matrix` (I guess a breaking changes, so maybe not in this PR... just a general idea)

File: src/Matrix.jl [hunk 3]
```diff
@@ -6896,14 +6910,37 @@ function identity_matrix(M::MatElem{T}, n::Int) where T <: NCRingElement
    z
 end
 
+################################################################################
+#
+#   Scalar matrix
+#
+################################################################################
+
+export scalar_matrix
```
Comment (lgoettgens) [inline src/Matrix.jl:6919]: Self-note: The exports needs to be moved to the top

File: src/julia/Matrix.jl (hunk 1 above)
Comment (lgoettgens) [reply src/julia/Matrix.jl:13]: Probably. But finding this is most time a bit tedious. I can have a look in there

File: src/julia/Matrix.jl (hunk 2 above)
Comment (lgoettgens) [reply src/julia/Matrix.jl:59]: The `Matrix` version is right above this one

File: src/Matrix.jl [hunk 4]
```diff
@@ -6817,6 +6817,26 @@ function matrix(R::NCRing, arr::AbstractMatrix{T}) where {T}
    end
 end
 
+function matrix(arr::AbstractMatrix{T}) where {T<:NCRingElem}
+   r, c = size(arr)
+   (r < 0 || c < 0) && error("Array must be non-empty")
+   R = parent(arr[1, 1])
+   all(e -> parent(e) === R, arr) || errpr("Non-compatible elements")
```
Comment (fingolfin) [inline src/Matrix.jl:6824]: Typo. So clearly this case is never tested ;-)
```suggestion
   all(e -> parent(e) === R, arr) || error("Non-compatible elements")
```

File: src/julia/Matrix.jl (hunk 1 above)
Comment (fingolfin) [reply src/julia/Matrix.jl:13]: Thank you!

File: src/Matrix.jl (hunk 4 above)
Comment (lgoettgens) [reply src/Matrix.jl]: Uff :)

**Review by fingolfin — APPROVED** [review]

**Comment by thofma** [comment]
There is now a conflict. I am not sure whether the method should be removed or not.

**Comment by lgoettgens** [comment]
I'll look at it

**Comment by lgoettgens** [comment]
> There is now a conflict. I am not sure whether the method should be removed or not.

The merge was non-trivial as the moved function has been changed in #1443. It should be fixed now.

**Pull request merged**